### PR TITLE
feat: Rewrite `isAdmin` mixin to a composable

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

This PR will replace the usage of `adminMixin` from the code base with the `useAdmin` composable.

Fixes https://linear.app/chatwoot/issue/CW-3448/rewrite-isadmin-mixin-to-a-composable

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please refer to this issue description.
https://linear.app/chatwoot/issue/CW-3448/rewrite-isadmin-mixin-to-a-composable#368d7900-562a-4b50-8e9d-f286e6675266

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by iamsivin** [comment]
Only merge this PR after merging this [feat: Replace `alertMixin` usage with `useAlert`](https://github.com/chatwoot/chatwoot/pull/9793) PR. Raised PR like this to solve the merge conflicts. The PR #9817 has a lot of file changes.

**Review by scmmishra — APPROVED** [review]

**Pull request merged**
